src/util/string_utils.cpp

```diff
@@ -156,13 +156,13 @@ std::string escape_non_alnum(const std::string &to_escape)
   std::ostringstream escaped;
   for(auto &ch : to_escape)
   {
+    const int uch{static_cast<unsigned char>(ch)};
```

Comment:
Nit: This is now complicated enough to warrant an explanation I'd think. Would be great if you could drop a line here of why this is done.

Reply:
Done.